common/persistence/src/main/java/app/coronawarn/server/common/persistence/service/utils/checkins/FakeCheckinsGenerator.java

```diff
@@ -0,0 +1,50 @@
+package app.coronawarn.server.common.persistence.service.utils.checkins;
+
+import static app.coronawarn.server.common.persistence.service.utils.checkins.FakeCheckinIntervalSpecification.*;
+
+import app.coronawarn.server.common.protocols.internal.pt.CheckIn;
+import com.google.protobuf.ByteString;
+import java.security.SecureRandom;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.springframework.stereotype.Component;
+
+@Component
+public class FakeCheckinsGenerator {
```

Comment:
Strictly speaking, fake checkins are application logic and not "persistence", IMHO. Would prefer to have anything related to fake checkins moved to the submission module (see [SubmissionController](https://github.com/corona-warn-app/cwa-server/blob/f06b7d2b0091f9fb9f6342238e31854597798fa2/services/submission/src/main/java/app/coronawarn/server/services/submission/controller/SubmissionController.java#L176) for reference).

Reply:
Agree to refactor this in a follow-up PR. @FelixRottler if you have created a backlog item  please link it here

Reply:
https://jira-ibs.wbs.net.sap/browse/EXPOSUREBACK-1091